### ISSUE TITLE
2.7 RC1 - unable to update an existing Pod's name

### ISSUE THREAD

**Issue by lougreenwood** [pods-framework/pods#4479]
When using 2.7 RC1, it seems it's not possible to edit the name of an existing pod:

Steps to reproduce
* Go to a 'Edit Pod' page for an existing pod in WP Admin
* Attempt to update the Pod's name
* See error dialog `Error: Unable to process request, please try again.`
* In inspector, check php error returned from Ajax request is:
`Undefined offset: 0 in file /usr/share/nginx/html/htdocs/content/plugins/pods/classes/PodsAPI.php on line 5491`

**Comment by jimtrue** [comment]
I'm not able to recreate this issue. What's the name of the pod you're changing from and to? And can you replicate on other pods in the same installation?

**Comment by lougreenwood** [comment]
Yep, it’s replicable on other Pods too.

The old name was ‘articles’ the new name is ‘articleskjlfdsh’ (or any random key-smush :D)

If I switch back to Pods 2.6.11, the issue goes away - if I drop 2.7 RC1 in, the issue comes back…

**Comment by jimtrue** [comment]
I just tested on two other local installations and I still can't replicate the problem on 2.7.0-rc-1

Going to kick this one up to @sc0ttkclark and @pglewis for confirmation, though.

**Comment by sc0ttkclark** [comment]
@jimtrue turn on WP_DEBUG when testing

**Comment by sc0ttkclark** [comment]
Not sure if the error here is caused by the php warning coming up, but this is the block of code:

```
                $pod = get_posts( array(
                    'name' => $params->name,
                    'post_type' => '_pods_pod',
                    'posts_per_page' => 1
                ) );

                if ( is_array( $pod ) ) {
                    $pod = $pod[0];
                }
```

**Comment by sc0ttkclark** [comment]
I've got a fix in #4480, can you confirm that resolves your issue @lougreenwood

**Comment by lougreenwood** [comment]
Yep, works for me - thanks!
